src/shogun/labels/MulticlassLabels.cpp

```diff
@@ -193,17 +193,13 @@ namespace shogun
 	{
 		auto int_labels = orig->get_int_labels();
 		std::set<int32_t> unique(int_labels.begin(), int_labels.end());
-		// TODO: anything discrete should be a possible multiclasslabels.
-		// Once that is the case, this check can be removed
-		// For now, if we don't enforce this, there will be crashes as the
-		// class is used to index vectors
-		REQUIRE(
-		    (*std::min_element(unique.begin(), unique.end())) == 0 &&
-		        (*std::max_element(unique.begin(), unique.end())) ==
-		            (index_t)unique.size() - 1,
-		    "Multiclass labels must be contiguous integers in [0, ..., "
-		    "num_classes -1].\n");
-
+		// anything discrete should be a possible multiclasslabels.
```

Comment:
few comments here:

* if there is no conversio needed, then no conversion should be done (i.e. leave the check in and only convert if that is false)
* pls do only copy the label vector **once**, currently I think you are copying things multiple times (your loop, `set_int_labels`, ...)
* why do you call `set_int_labels`? I would rather just convert the label vector in here and then pass that vector itself to `some<CMulticlassLabels>`, rather than modifying the given label
* we might in fact want to make the passed `orig` pointer const to avoid changing the original structure ....

Reply:
actually, I am just working on this anyways, so let me do the fix and then you can use my code....stay tuned

Reply:
#4251